### PR TITLE
chore: simplify `noFormat`

### PR DESCRIPTION
**Pull request by fisker**
## PR Checklist

- [ ] Addresses an existing open issue: fixes #000
- [ ] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [ ] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

### CODE REVIEW

**Review by JoshuaKGoldberg — APPROVED** [review]
Ha, TIL - nice. Thanks!

File: packages/utils/src/eslint-utils/rule-tester/RuleTester.ts [hunk 1]
```diff
@@ -306,12 +306,8 @@ class RuleTester extends BaseRuleTester.RuleTester {
...
  *   for the internal/plugin-test-formatting lint rule
  */
-function noFormat(strings: TemplateStringsArray, ...keys: string[]): string {
-  const lastIndex = strings.length - 1;
-  return (
-    strings.slice(0, lastIndex).reduce((p, s, i) => p + s + keys[i], '') +
-    strings[lastIndex]
-  );
+function noFormat(raw: TemplateStringsArray, ...keys: string[]): string {
+  return String.raw({ raw }, ...keys);
```
Comment (Josh-Cena) [inline packages/utils/src/eslint-utils/rule-tester/RuleTester.ts:310]: This won't work if there's any escape sequence in the string. See  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/raw#building_an_identity_tag. I'm not sure if this is acceptable, considering we may have escape sequences in the test cases (e.g. within string literals)

Comment (fisker) [reply packages/utils/src/eslint-utils/rule-tester/RuleTester.ts:310]: This function mean to use cooked value,not raw.

Comment (Josh-Cena) [reply packages/utils/src/eslint-utils/rule-tester/RuleTester.ts:310]: Ah, sorry I misread.

**Pull request merged**